File: lambdaconscarcdr.erl
Kons  = fun(Hd, Tl) ->
  fun
    (true ) -> Hd;
    (false) -> Tl
  end
end,

Virst = fun(L) -> L(true ),
Rrest = fun(L) -> L(false),

Lstt  = Kons(1, Kons(2, nil)),

Virst(Rrest(Lstt)).

File: lambdaconscarcdr.py
#!/usr/bin/env python

kons  = lambda hd, tl: lambda x: hd if x else tl
virst = lambda l: l(True )
rrest = lambda l: l(False)

lstt  = kons(1, kons(2, None))

virst(rrest(lstt))
